### DISCUSSION TITLE
Github actions platform flags

### DISCUSSION THREAD

**Discussion by thatTPR** [community/community#6536] [Actions]
Use systeminfo apis to retrieve platform specific information and retrieve action results with platform flags. I know the entire codespace thing is supposed to be platform agnostic but reality is a bit different and extra tooling can't hurt.

**Comment by asciimike** [comment]
@thatTPR interesting request, what are you hoping to do with this? For instance, are you building for certain architectures or relying on specific instructions to exist?

**Reply by thatTPR** [reply]
Not as such but I guess this could be used for this as well. Some build procedures are platform dependent and I thought it would be nice to integrate that into a platform independent tool like GH actions. So let's say I need clang on Linux but msvc on windows for a certain repo to build. GH actions could handle that. And that way I could just run the action locally and not have to worry about platform specific tooling. Would be great for reproducibility . I haven't used the tool much so I am not exactly sure if there is not a way to do this already. But what I am certain does not exist is specific platform flags like RAM GPU CPU , arch etc. That way the more the software is developed and merged into the main repo. The more configurations are supported and some of the guesswork would be gone. It is not hat useful for software devs. But might be useful for hardware devs if they are targeting a framework like OpenCl or CUDA
